token/core/zkatdlog/nogh/issue.go

```diff
@@ -98,8 +139,24 @@ func (s *IssueService) Issue(ctx context.Context, issuerIdentity driver.Identity
 		return nil, nil, err
 	}
 
+	// prepare inputs to redeem, if any
+	if opts != nil && opts.TokenConversionRequest != nil {
+		tokens := opts.TokenConversionRequest.Tokens
+		if len(tokens) != 0 {
```

Comment:
to avoid nesting, we could add this condition to the previous clause
```
if opts != nil && opts.TokenConversionRequest != nil && len(opts.TokenConversionRequest.Tokens) > 0
```

Reply:
okay